### PR TITLE
work suspension: webhook validation

### PR DESCRIPTION
**Pull request by a7i**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #5118

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

**Comment by a7i** [comment]
/hold

until https://github.com/karmada-io/karmada/pull/4838 is merged

**Comment by XiShanYongYe-Chang** [comment]
Hi @a7i we can continue this task ^-^

**Review by RainbowMango — COMMENTED** [review]
/assign

File: pkg/util/validation/validation.go [hunk 1]
```diff
@@ -55,18 +56,37 @@ func validateResourceSelectorsIfPreemptionEnabled(spec policyv1alpha1.Propagatio
...
+		return nil
+	}
+
+	if suspension.DispatchingOnClusters == nil || len(suspension.DispatchingOnClusters.ClusterNames) == 0 {
+		return nil
+	}
+
+	return field.ErrorList{
+		field.Invalid(fldPath.Child("suspension"), suspension, "suspension dispatching cannot co-exist with dispatchingOnClusters.clusterNames"),
+	}
```
Comment (RainbowMango) [inline pkg/util/validation/validation.go:81]: I would prefer this straightforward style:
```suggestion
	if (suspension.Dispatching != nil && *suspension.Dispatching == true) &&
		(suspension.DispatchingOnClusters != nil && len(suspension.DispatchingOnClusters.ClusterNames) > 0) {

		return field.ErrorList{
			field.Invalid(fldPath.Child("suspension"), suspension, "suspension dispatching cannot co-exist with dispatchingOnClusters.clusterNames"),
		}
	}

	return nil
```

**Review by RainbowMango — COMMENTED** [review]
Generally looks good.

I'm surprised that this PR is not based on the latest commit, but all tests are passing. I fetched the PR on my side (by `git fetch upstream pull/5282/head:5282`) and it failed to compile as missing the API definition.

I'm unsure if this is due to GitHub Action changing [how it fetches code](https://github.com/karmada-io/karmada/actions/runs/10270480293/job/28418444032?pr=5282#step:2:175). 

Anyway, @a7i could you please help to rebase this and push again?

**Comment by a7i** [comment]
thanks for the review @RainbowMango 

rebased and applied the suggestion

**Review by RainbowMango — APPROVED** [review]
/lgtm
/approve

**Comment by RainbowMango** [comment]
/retest
The failing case seems unrelated.

File: pkg/util/validation/validation.go (hunk 1 above)
Comment (grosser) [reply pkg/util/validation/validation.go]: the old version was easier to digest for me ... ideally 1 comment per `if` explaining what is going on
suggested feels a overwhelming

Comment (RainbowMango) [reply pkg/util/validation/validation.go]: I'm sorry for not explaining why I'm suggesting that in detail.

Firstly the old version is great, it looks simple. But given that we need a single check on whether the two fields co-exist, I tend to a more cohesive approach to make it easier to add more checks in the future.

**Pull request merged**